crates/api/src/sitemap.rs

```diff
@@ -99,45 +95,47 @@ pub(crate) mod tests {
       root
         .children()
         .next()
-        .unwrap()
+        .ok_or(LemmyErrorType::NotFound)?
         .children()
         .find(|element| element.tag().name() == "loc")
-        .unwrap()
+        .ok_or(LemmyErrorType::NotFound)?
         .text(),
       "https://example.com/"
     );
     assert_eq!(
       root
         .children()
         .next()
-        .unwrap()
+        .ok_or(LemmyErrorType::NotFound)?
         .children()
         .find(|element| element.tag().name() == "lastmod")
-        .unwrap()
+        .ok_or(LemmyErrorType::NotFound)?
         .text(),
       "2022-12-01T09:10:11+00:00"
     );
     assert_eq!(
       root
         .children()
         .nth(1)
-        .unwrap()
+        .ok_or(LemmyErrorType::NotFound)?
         .children()
         .find(|element| element.tag().name() == "loc")
-        .unwrap()
+        .ok_or(LemmyErrorType::NotFound)?
         .text(),
```

Comment:
```rust

      root
        .children()
        .next()
        .map(|n| n.children().find(|element| element.tag().name() == "loc"))
        .flatten()
        .map(|c| c.text())
        .unwrap_or_default(),
```

Reply:
Clippy recommended using `and_then()` instead of `.map().flatten()`, so I went with that. 

```rs
.and_then(|n| n.children().find(|element| element.tag().name() == "loc"))
.map(Element::text)
```